src/uct/api/uct.h

```diff
@@ -2470,6 +2470,23 @@ ucs_status_t uct_md_config_read(uct_component_h component,
                                 uct_md_config_t **config_p);
 
 
+/**
+ * Apply key/val to uct_config
+ *
+ * @param config  uct config bundle start addr
+ * @param key&val key&val to be applied
+ */
+ucs_status_t uct_apply_config(void *config, const char *key, const char *val);
+
+
+/**
+ * Apply per key/val in list to uct_config
+ *
+ * @param config         uct config bundle start addr
+ * @param config_kv_list key/val list to be applied
+ */
+void uct_apply_config_list(void *config, ucs_list_link_t *config_kv_list);
```

Comment:
why need new API? we already have uct_config_modify

Reply:
@yosefe 
1. ```uct_config_modify``` support only one key/value pair to be applied to the ```config```
2. Here, all the cached key/value pairs need to be applied to the ```config```
3. ```uct_apply_config_list``` wrap ```uct_config_modify```

Reply:
API should not have 2 methods of doing same thing. We can call uct_config_modify in a list from UCP layer

Reply:
@yosefe I'll remove the wrap here. Note: This will increase lines of code since uct_apply_config_list is called in several places.

Reply:
@yosefe Done

Reply:
can move the function to UCP to reduce code lines

Reply:
@yosefe Do you mean that we could still wrap the ```uct_config_modify``` into ```uct_apply_config``` and ```uct_apply_config_list``` and move these two functions into ucp library?

Reply:
yes, exactly. the problem was with adding UCT API which is not really needed. but it's ok if it's an internal function in ucp. just need to rename them to ucp_*

Reply:
Done